symengine/ruby/ext/symengine/ruby_basic.c

```diff
@@ -5,23 +5,17 @@ void cbasic_free(void *ptr){
     basic_free(basic_ptr);
 }
 
-VALUE cbasic_alloc(VALUE klass){
-    VALUE obj;
-    basic_struct *struct_ptr;
-
-    obj = Data_Make_Struct(klass, basic_struct, NULL, cbasic_free, struct_ptr);
-
-    struct_ptr->data = NULL;
+void cbasic_free_heap(void *ptr) {
+    basic_struct *basic_ptr = ptr;
+    basic_free_heap(basic_ptr);
+}
 
-    return obj;
+VALUE cbasic_alloc(VALUE klass){
+    basic_struct *struct_ptr = basic_new_heap();
+    return Data_Wrap_Struct(klass, NULL, cbasic_free_heap, struct_ptr);
 }
 
 VALUE cbasic_init(VALUE self){
-    basic_struct *this;
-
-    Data_Get_Struct(self, basic_struct, this);
-    basic_init(this);
-
     return self;
 }
```

Comment:
I was wrong. It does get a default `initialize` method. Allocation on the heap is done by the method `alloc`. In ruby, the creation of objects happen in two stages, first it gets allocated(`alloc`) and the initialized(`initialize`). We already have a alloc_func for basic, so this can be removed. Thanks for pointing it out.